Assets/Mirror/Editor/MirrorAssetProcessor.cs

```diff
@@ -0,0 +1,113 @@
+using UnityEngine;
+using UnityEditor;
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace DigitalRuby.WeatherMaker
+{
+    /// <summary>
+    /// Pre-process define utility
+    /// </summary>
+    public class MirrorAssetPostprocessor : AssetPostprocessor
+    {
+        private const string assetToCheck = "/Mirror/Runtime/NetworkIdentity.cs";
+        private const string preProcessor = "MIRROR_NET";
+
+        /// <summary>
+        /// Add the pre-processor
+        /// </summary>
+        internal static void AddPreProcessor()
+        {
+            string currBuildSettings = UnityEditor.PlayerSettings.GetScriptingDefineSymbolsForGroup(UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup);
+            if (!currBuildSettings.Contains(preProcessor))
+            {
+                Debug.LogWarning("Updating preprocessor to " + currBuildSettings);
+                currBuildSettings += ";" + preProcessor;
+                UnityEditor.PlayerSettings.SetScriptingDefineSymbolsForGroup(UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup, currBuildSettings);
+            }
+        }
+
+        /// <summary>
+        /// Remove the pre-processor
+        /// </summary>
+        internal static void RemovePreProcessor()
+        {
+            string currBuildSettings = UnityEditor.PlayerSettings.GetScriptingDefineSymbolsForGroup(UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup);
+            int origCurrBuildSettingsLength = currBuildSettings.Length;
+            currBuildSettings = System.Text.RegularExpressions.Regex.Replace(currBuildSettings, ";?" + preProcessor + ";?", string.Empty);
+            if (currBuildSettings.Length != origCurrBuildSettingsLength)
+            {
+                Debug.LogWarning("Updating preprocessor to " + currBuildSettings);
+                UnityEditor.PlayerSettings.SetScriptingDefineSymbolsForGroup(UnityEditor.EditorUserBuildSettings.selectedBuildTargetGroup, currBuildSettings);
+            }
+        }
+
+        internal static void ProcessImportedAssets(params string[] importedAssets)
+        {
+            if (importedAssets == null)
+            {
+                return;
+            }
+
+            foreach (string assetName in importedAssets)
+            {
+                if (assetName.IndexOf(assetToCheck, System.StringComparison.OrdinalIgnoreCase) >= 0)
+                {
+                    AddPreProcessor();
+                }
+            }
+        }
+
+        internal static void ProcessDeletedAssets(params string[] deletedAssets)
+        {
+            if (deletedAssets == null)
```

Comment:
can this happen?

Reply:
I'd like to think Unity will never pass a null array, but it is possible. https://stackoverflow.com/questions/6584131/c-sharp-is-it-possible-to-have-null-params